be/src/exec/vectorized/chunks_sorter_full_sort.h

```diff
@@ -28,27 +29,26 @@ class ChunksSorterFullSort : public ChunksSorter {
     Status update(RuntimeState* state, const ChunkPtr& chunk) override;
     Status done(RuntimeState* state) override;
     void get_next(ChunkPtr* chunk, bool* eos) override;
-    DataSegment* get_result_data_segment() override;
-    uint64_t get_partition_rows() const override;
-    Permutation* get_permutation() const override;
-
     bool pull_chunk(ChunkPtr* chunk) override;
 
-    int64_t mem_usage() const override;
+    SortedRuns get_sorted_runs() override;
+    size_t get_output_rows() const override;
 
-    friend class SortHelper;
+    int64_t mem_usage() const override;
 
 private:
-    Status _sort_chunks(RuntimeState* state);
-    Status _build_sorting_data(RuntimeState* state);
-    Status _sort_by_column_inc(RuntimeState* state);
-
-    void _append_rows_to_chunk(Chunk* dest, Chunk* src, const Permutation& permutation, size_t offset, size_t count);
-
-    ChunkUniquePtr _big_chunk;
-    std::unique_ptr<DataSegment> _sorted_segment;
-    mutable Permutation _sorted_permutation;
-    std::vector<uint32_t> _selective_values; // for appending selective values to sorted rows
+    Status _merge_unsorted(RuntimeState* state, const ChunkPtr& chunk);
```

Comment:
would better the order strategy is:

1. _merge_unsorted
2. _partial_sort
3. _merge_sorted

Reply:
done